src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY = 16;
+    private Node<K, V>[] nodes;
+    private int size;
+
+    public MyHashMap() {
+        nodes = new Node[CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        thresholdCheck();
+        Node<K, V> previous = nodes[getHash(key)];
+        if (previous == null) {
+            nodes[getHash(key)] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        while (true) {
+            if (Objects.equals(previous.key, key)) {
+                previous.value = value;
+                return;
+            }
+            if (previous.next == null) {
+                previous.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
```

Comment:
after that you can take this part of the code out of the loop "while" and remove `if (previous.next == null) {`